boards/raspberry_pi_pico/README.md

```diff
@@ -0,0 +1,76 @@
+Raspberry Pi Pico - RP2040
+==========================
+
+<img src="https://www.raspberrypi.org/homepage-9df4b/static/bdc42b00ebe8f2312c4d229beb9325bf/53263/7d247ace-afb2-4555-b7b3-4f236eb779d6_Raspberry%2BPi%2BPico%2B1.jpg" width="35%">
+
+The [Raspberry Pi Pico](https://www.raspberrypi.org/products/raspberry-pi-pico/) is a
+board developed by the Raspberry Pi Foundation and is based on the RP2040 chip.
+
+## Getting Started
+
+First, follow the [Tock Getting Started guide](../../../doc/Getting_Started.md)
+
+## Flashing the kernel
+
+The kernel can be programmed using a second Raspberry Pi device and OpenOCD. OpenOCD needs to be installed on the Regular Raspberry Pi device.
+
+### Raspberry Pi Setup
+
+To install OpenOCD on the Raspberry Pi run the following commands on the Pi:
+```bash
+$ sudo apt-get update
+$ sudo apt install automake autoconf build-essential texinfo libtool libftdi-dev libusb-1.0-0-dev git
+$ git clone https://github.com/raspberrypi/openocd.git --recursive --branch rp2040 --depth=1
+$ cd openocd
+$ ./bootstrap
+$ ./configure --enable-ftdi --enable-sysfsgpio --enable-bcm2835gpio
+$ make -j4
+$ sudo make install
+$ cd ~
+```
+
+Enable SSH on the Raspberry Pi by following the [instructions on the Raspberry Pi website](https://www.raspberrypi.org/documentation/remote-access/ssh/).
+
+Next, connect the Pico to the Raspberry Pi by following the instructions in the [official documentation](https://datasheets.raspberrypi.org/pico/getting-started-with-pico.pdf#%5B%7B%22num%22%3A22%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C115%2C431.757%2Cnull%5D).
+### Flash the tock kernel
+
+`cd` into `boards/raspberry_pi_pico` directory and run:
+
+```bash
+$ make
+
+(or)
+
+$ make debug
+```
+
+Connect via ssh to the Raspberry Pi and forward port 3333. Then start OpenOCD on the Pi.
+```bash
+$ ssh pi@<pi_IP> -L 3333:localhost:3333
+
+(wait to connect)
+
+$ openocd -f interface/raspberrypi-swd.cfg -f target/rp2040.cfg
+```
+You can also open a serial console on the Raspberry Pi for debug messages.
+```bash
+$ sudo apt install minicom
+$ minicom -b 115200 -o -D /dev/serial0
+```
+
```

Comment:
At least openocd has to run on the Pi. Of course, using a serial TTL adapter works for using the serial port on the computer. But since most users will have a Raspberry Pi, this was an easier way, by simply adding three cables.

Reply:
done